### ISSUE TITLE
API-42: MicroProfile Rest Client 1.3 Epic

### ISSUE THREAD

**Issue by andymc12** [OpenLiberty/open-liberty#7583]
Epic for [API-42](https://bigblue.aha.io/features/API-42) feature.  
WAD available here: https://ibm.ent.box.com/file/455061283606

MP Rest Client 1.3.1 release page: https://github.com/eclipse/microprofile-rest-client/releases/tag/1.3.1

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## List of Steps to complete or get approvals / sign-offs for Onboarding to the Liberty release (GM date)
 
Instructions:
 - Do the actions below and mark them complete in the checklist when they are done.
 - Make sure all feature readiness approvers put the appropriate tag on the epic to indicate their approval.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **Before Development Starts or 8 weeks before Onboarding** 
- [x] POC Design / WAD Review Scheduled (David Chang) or N/A. AHA Feature marked.
- [x] POC Design / WAD Reviewed (Feature Owner) or N/A. AHA Feature marked.
- [x] Complete any To-Do's from the POC Review, and create a new TO-DO in the AHA Feature for the "Design Approval Request" and assign it to Ian Robinson. Leave the date blank. (Feature Owner)
- [x] Design / WAD Approval (Ian Robinson) or N/A. AHA Feature marked.
- [ ] No Design / No WAD Approval (Arthur De Magalhaes - cloud / Alasdair Nottingham - server) or *N/A*. AHA Feature marked.
- [x] AHA Feature has link to Github epic or issue. Github epic has link back to AHA Feature. (Feature Owner or Backlog Subtribe PM).
- [x] SVT Requirements identified. (Epic owner / Feature owner with SVT focal point)
- [x] ID Requirements identified. (Epic owner / Feature owner with ID focal point)
- [x] Create a child task of the epic entitled "FAT Approval Test Summary"
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **3 weeks before Onboarding** 
- [x] Identify all open source libraries that are changing or are new. Work with Legal Release Services (Cass Tucker or Release PM) to get open source cleared and approved. Or N/A. (Epic Owner).   New or changed open source impacts license and Certificate of Originality.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE ** 3 weeks before Onboarding** 
- [x] All new or changed PII messages are checked into the integration branch, before the last translation shipment out. (Epic Owner)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **2 weeks before Onboarding** 
- [x] Implementation complete. (Epic owner / Feature owner) 
- [x] All function tests complete. Ready for FAT Approval. (Epic owner / Feature owner)
- [x] Review all known issues for Stop Ship. (Epic owner / Feature owner / PM)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## APPROVALS with TARGET COMPLETION DATE **2 to 1 week before Onboarding** 
Prereq: You must have the Design Approval or No Design Approval on the AHA Feature.
- [x] **Accessibility** - (G Scott Johnston). Accessibility testing is complete or N/A. Approver adds label focalApproved:accessibility to the Epic in Github.
- [x] **FAT Liberty SOE** - (Kevin Smith). SOE FATS are running successfully or N/A . Approver adds label focalApproved:fat to the Epic in Github.
- [x] **Globalization** (Marika Joannidis - Liberty / Simy Cheeran - tWAS). Translation is complete or N/A. TVT - complete or N/A. Approver adds label focalApproved:globalization to the Epic in Github.
- [x] **ID** - (Kareen Deen). Documentation work is complete or N/A . Approver adds label focalApproved:id to the Epic in Github.
- [x] **Performance** - (Jared Anderson). Performance testing is complete with no high severity defects or N/A . Approver adds label focalApproved:performance to the Epic in Github.
- [x] **Serviceability** - (Don Bourne). Serviceability has been addressed. 
- [x] **STE** - (Swati Kasundra). STE chart deck is complete or N/A . Approver adds label focalApproved:ste to the Epic in Github.
- [x] **SVT** - (Greg Ecock - Cloud, Brian Hanczaryk- APS). SVT is complete or N/A . Approver adds label focalApproved:svt to the Epic in Github.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## TARGET COMPLETION DATE **1 week before Onboarding** 
- [x] No Stop Ship issues for the feature. (Epic owner / Feature owner / Release PM)
- [ ] Ship Readiness Review and Release Notes completed (Epic owner / Feature owner / Release PM)
- [ ] Github Epic and Epic's issues are closed / complete. All PRs are committed to the master branch. (Epic owner / Feature owner / Backlog Subtribe PM)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## NOT REQUIRED FOR A FEATURE
- [ ] **OL Guides** - (Yee-Kang Chang). Assessment for OL Guides is complete or N/A. 
- [ ] **WDT** - (Leonard Theivendra). WDT work complete or N/A.
 
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## Related Deliverables TARGET COMPLETION DATE **General Availability** 
- [ ] Blog article writeup (Epic owner / Feature owner / Laura Cowen)

**Comment by marikaj123** [comment]
According to Andy, the feature is not waiting for new files that need a translation.

**Comment by gscottj** [comment]
This feature does not add any new UI or change any existing UI in ways that require accessibility verification testing.

**Comment by chirp1** [comment]
From Andy, this feature will be available in the 19007 time frame and has no additional ID. Approving.

**Comment by marikaj123** [comment]
@andymc12 - Please add the label WAD or No-WAD approval

**Comment by skasund** [comment]
No STE is needed for this feature

**Comment by dave-waddling** [comment]
**FAT Focal Approval comment:** Please can you complete the Feature Test Summary issue (#7597) and also complete the WAD Approval process then add either the "WAD Approved" or "NoWAD Approved" labels.

Once you have delivered all of your code/testing, completed the Feature Test Summary, and gotten the WAD approval please add the "FAT Complete" label to trigger a review. Thanks!

**Comment by donbourne** [comment]
**Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated? 
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

4.  Which L2 / L3 queues will handle PMRs for this feature?  Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it.  Ask Don Bourne if you need links or more info.

**Comment by marikaj123** [comment]
@andymc12 - Please ensure not to move the feature t GA before receiving all of the appropriate focal point approvals.

**Comment by dave-waddling** [comment]
**FAT Focal Approval comment:** Hi @andymc12, just letting you know I took a look to see if this feature is ready for review. It looks like the issue is currently not "FAT Complete" and the Feature Test Summary (#7597) is incomplete. Once these issues are rectified I'll be able to review.

**Comment by andymc12** [comment]
I chatted with @dave-waddling and @garypicher about this issue.  The holdup is the SSL TCK tests that fail on IBM JDK 8 (all other tests pass 100%).  This issue is documented internally at defect [264153](https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.workitem.viewWorkItem&id=264153) and in the MP Rest Client project at issue [204](https://github.com/eclipse/microprofile-rest-client/issues/204). 

The SSL TCK has been updated and released ([1.3.3](https://github.com/eclipse/microprofile-rest-client/releases/tag/1.3.3)), but we are currently waiting to pull in that release (see PR #8147) and re-run the SOEs with IBM JDK 8.  Since this will take place after we hope to ship the MP Rest Client 1.3 feature, Dave and Gary have approved an SOE exception.  We will still need the approval, but that should come next week after running the SOE with newer TCK release.  I have verified locally that the tests pass with the newer TCK on Mac and Linux - both running IBM JDK 8.

**Comment by dave-waddling** [comment]
**FAT Focal Approval comment:** I've reviewed the Feature Test Summary and, given #8147 demonstrates the TCK fix was successful, I'm happy to add the FAT Focal Approval.

**Comment by marikaj123** [comment]
@andymc12 - Was the feature code enabled and included into the Liberty 19007 GM build?

**Comment by tevans78** [comment]
Yes it was delivered ...  I think this Epic is complete?

**Comment by marikaj123** [comment]
Yes, please close the Epic.

**Comment by atosak** [comment]
Andy confirmed we can close this issue.
